File: FPG.Core/Factories/HashingAlgorithmFactory.cs
using System;

using FPG.Algorithms;
using FPG.Interfaces;

namespace FPG.Factories
{
    public static class HashingAlgorithmFactory
    {
        public static IHashingAlgorithm Create(String name)
        {
            switch (name.ToLower())
            {
                case "sha512":
                    return new SHA512HashingAlgorithm();
                default:
                    throw new ArgumentException("Unknown algorithm name");
            };
        }
    }
}

File: lib/core/factories/HashingAlgorithmFactory.dart
import 'package:fpg/core/algorithms/hashing/SHA512HashingAlgorithm.dart';
import 'package:fpg/core/interfaces/IHashingAlgorithm.dart';

class HashingAlgorithmFactory {
  static IHashingAlgorithm create(String name) {
    switch (name.toLowerCase()) {
      case "sha512":
        return SHA512HashingAlgorithm();
      default:
        throw FormatException("Unknown algorithm name");
    }
  }
}
